Coverage/Main.cpp

```diff
@@ -122,22 +119,75 @@ void ParseCommandLine(int argc, const char **argv)
 		}
 		else
 		{
-			throw std::exception("Incorrect parameter.");
+            std::string message("Incorrect parameter: ");
+            message += s;
+			throw std::exception(message.c_str());
 		}
 	}
+    auto idx = cmdLine.find(" -- ");
+    if (idx == std::string::npos)
+    {
+        throw std::exception("Expected executable filename in command line.");
+    }
+
+    idx += 2;
+    while (idx < cmdLine.size() && cmdLine[idx + 1] == ' ') { ++idx; }
+
+    std::string childCommand = cmdLine.substr(idx+1);
+    if (childCommand.size() == 0)
+    {
+        throw std::exception("Expected executable filename in command line.");
+    }
+
+    opts.ExecutableArguments = childCommand;
+    /*
+    size_t pos = opts.ExecutableArguments.find(opts.Executable);
+    opts.ExecutableArguments = opts.ExecutableArguments.substr( pos + opts.Executable.size());
+    if(opts.ExecutableArguments[0] == '\"')
+        opts.ExecutableArguments = opts.ExecutableArguments.substr( std::min<size_t>(2, opts.ExecutableArguments.size()) );
+    else if(!opts.ExecutableArguments.empty())
+        opts.ExecutableArguments = opts.ExecutableArguments.substr(1);
+    */
+#ifdef _DEBUG
```

Comment:
I seriously vote against using #ifdef _DEBUG use in programs, because they have the tendency to end up all over the place. That said, I reckon the fact that I did way too little with proper debug messages in the solution.

IMO it's way better to use a variadic template for this. These are easy to remove in release builds using some compile-time trickery or a preprocessor directive, and more flexible in the long run. E.g.

template <typename... Args>
inline void Message(int level, const std::string& str, Args... args) { // expand template and push to output }